src/InMemoryMessageRepository.php

```diff
@@ -44,12 +44,29 @@ public function persist(Message ...$messages)
 
     public function retrieveAll(AggregateRootId $id): Generator
     {
+        $lastMessage = null;
+
+        foreach ($this->messages as $message) {
+            if ($id->toString() === $message->header(Header::AGGREGATE_ROOT_ID)->toString()) {
+                yield $message;
+                $lastMessage = $message;
+            }
+        }
+
+        return $lastMessage instanceof Message ? $lastMessage->aggregateVersion() : 0;
+    }
+
+    public function retrieveAllAfterVersion(AggregateRootId $id, int $aggregateRootVersion): Generator
+    {
+        $lastMessage = null;
+
         foreach ($this->messages as $message) {
             if ($id->toString() === $message->header(Header::AGGREGATE_ROOT_ID)->toString()) {
```

Comment:
Shouldn't this check for `> $aggregateRootVersion`?

Reply:
True! Fixed.